src/parser/jobs/gnb/modules/AmmoCombo.tsx

```diff
@@ -0,0 +1,262 @@
+import {BuffEvent, CastEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {Timeline} from 'parser/core/modules/Timeline'
+
+const RELEVANT_ACTIONS = [ACTIONS.JUGULAR_RIP.id, ACTIONS.ABDOMEN_TEAR.id, ACTIONS.EYE_GOUGE.id]
+const RELEVANT_STATUSES = [STATUSES.READY_TO_RIP.id, STATUSES.READY_TO_TEAR.id, STATUSES.READY_TO_GOUGE.id]
+const COMBO_BREAKERS = [ACTIONS.KEEN_EDGE.id, ACTIONS.BRUTAL_SHELL.id, ACTIONS.SOLID_BARREL.id] // These skills will break the Gnashing combo
+const COMBO_ACTIONS = [ACTIONS.GNASHING_FANG.id, ACTIONS.JUGULAR_RIP.id, ACTIONS.SAVAGE_CLAW.id, ACTIONS.ABDOMEN_TEAR.id, ACTIONS.WICKED_TALON.id, ACTIONS.EYE_GOUGE.id]
+
+// Expected within a standard gnashing.
+const EXPECTED_USES = {
+	JUGULAR_RIP: 1,
+	SAVAGE_CLAW: 1,
+	ABDOMEN_TEAR: 1,
+	WICKED_TALON: 1,
+	EYE_GOUGE: 1,
+}
+
+const GNASHING_TIME = 7500 // time in ms, this is the "fastest" time a combo can happen, but can take longer TODO: Dropped Combos?
+
+class GnashingComboState {
+	start: number
+	end?: number
+	rotation: CastEvent[] = []
+	// isRushing: boolean = false //I've yet to figure out how I wanted to handle end of fights rushes
+	isProper: boolean = false
+
+	// the trackers
+	numGnashing: number = 0
+	numJugular: number = 0
+	numSavage: number = 0
+	numAbdomen: number = 0
+	numWicked: number = 0
+	numEye: number = 0
+	numTotal: number = 0 // This is the filter, sub 6 = possible problem
+
+	constructor(start: number) {
+		this.start = start
+	}
+}
+
+export default class AmmoCombo extends Module {
+	static handle = 'Gnashing Fang Combo issues'
+
+	@dependency private checklist!: Checklist
+	@dependency private timeline!: Timeline
+
+	private buffs = 0
+	private actions = 0
+
+	private gnashingComboWindows: GnashingComboState[] = [] // Store all the gnashing fang combos to be output to the rotationTable
+
+	private get lastGnashingCombo(): GnashingComboState | undefined {
+		return _.last(this.gnashingComboWindows)
+	}
+
+	protected init() { // TODO: Strip away the other hooks? I could roll them into the actual cast I have for tracking the combo, but won't be as nice looking IMO
+		this.addHook('cast',
+			{
+				by: 'player',
+				abilityId: RELEVANT_ACTIONS,
+			},
+			() => this.actions++)
+
+		this.addHook('cast', {by: 'player'}, this.onCast)
+
+		this.addHook('applybuff',
+			{
+				by: 'player',
+				abilityId: RELEVANT_STATUSES,
+			},
+			() => this.buffs++)
+		this.addHook('complete', this.onComplete)
+	}
+
+	private onCast(event: CastEvent) {
+		const actionId = event.ability.guid
+
+		// cases to ignore, No Mercy any other modules will handle the latter 2.
+
+		if (actionId === ACTIONS.ATTACK.id || actionId === ACTIONS.BURST_STRIKE.id || actionId === ACTIONS.SONIC_BREAK.id ) {
+			return
+		}
+
+		if (actionId === ACTIONS.GNASHING_FANG.id) {
+
+			this.onEndGnashingCombo(event) // Make damn sure the previous window has ended.
```

Comment:
This is going to ALWAYS overwrite the end time for the previous Gnashing Fang combo, is that intended?  It looks like nothing's really happening with the window end time currently; is that planned for future handling or should the end time tracking just be removed altogether as unused?

Reply:
unintended override there.

end time is currently planned to be used in a custom use checker, but I'll discuss this more in discord with you, if I can make it work on the standard one, I'd prefer to use cores, but due to to GNB's unique cases I'm hesitating to use it.